lib/membrane/rtp/payload_format.ex

```diff
@@ -0,0 +1,128 @@
+    15 => %{encoding_name: :G728, clock_rate: 8000},
+    16 => %{encoding_name: :DVI4, clock_rate: 11025},
+    17 => %{encoding_name: :DVI4, clock_rate: 22050},
+    18 => %{encoding_name: :G729, clock_rate: 8000},
+    25 => %{encoding_name: :CELB, clock_rate: 90000},
+    26 => %{encoding_name: :JPEG, clock_rate: 90000},
+    28 => %{encoding_name: :NV, clock_rate: 90000},
+    31 => %{encoding_name: :H261, clock_rate: 90000},
+    32 => %{encoding_name: :MPV, clock_rate: 90000},
+    33 => %{encoding_name: :MP2T, clock_rate: 90000},
+    34 => %{encoding_name: :H263, clock_rate: 90000}
+  }
+
+  @enforce_keys [:encoding_name]
+  defstruct @enforce_keys ++ [payload_type: nil, payloader: nil, depayloader: nil]
+
+  @type t :: %__MODULE__{
+          encoding_name: RTP.encoding_name_t(),
+          payload_type: RTP.payload_type_t() | nil,
+          payloader: module | nil,
+          depayloader: module | nil
+        }
+
+  @doc false
+  @spec register_static_formats() :: :ok
+  def register_static_formats() do
+    @payload_types_specs
+    |> Enum.group_by(fn {_pt, specs} -> specs.encoding_name end, fn {pt, _specs} -> pt end)
+    |> Enum.each(fn
+      {name, [pt]} -> register(%__MODULE__{encoding_name: name, payload_type: pt})
+      _ambiguous -> :ok
+    end)
+  end
+
+  @doc """
+  Returns encoding name and clock rate for given payload type, if registered.
+  """
+  @spec get_payload_type_mapping(RTP.payload_type_t()) :: %{
+          optional(:encoding_name) => RTP.encoding_name_t(),
+          optional(:clock_rate) => RTP.clock_rate_t()
+        }
+  def get_payload_type_mapping(payload_type) when RTP.is_payload_type_static(payload_type) do
+    Map.fetch!(@payload_types_specs, payload_type)
+  end
+
+  def get_payload_type_mapping(payload_type) when RTP.is_payload_type_dynamic(payload_type) do
+    Application.get_env(@app, {:payload_type, payload_type}, %{})
+  end
+
+  @doc """
+  Registers default encoding name and clock rate for a dynamic payload_type
+  """
+  @spec register_payload_type_mapping(
+          RTP.dynamic_payload_type_t(),
+          RTP.encoding_name_t(),
+          RTP.clock_rate_t()
+        ) :: :ok | no_return()
+  def register_payload_type_mapping(payload_type, encoding_name, clock_rate)
+      when RTP.is_payload_type_dynamic(payload_type) do
+    case Application.fetch_env(@app, {:payload_type, payload_type}) do
+      {:ok, payload_format} ->
+        raise "RTP payload type #{payload_type} already registered: #{inspect(payload_format)}"
+
+      :error ->
+        Application.put_env(@app, {:payload_type, payload_type}, %{
+          encoding_name: encoding_name,
+          clock_rate: clock_rate
+        })
+    end
+  end
+
+  @doc """
+  Returns payload format registered for given encoding name.
+  """
+  @spec get(RTP.encoding_name_t()) :: t
+  def get(encoding_name) do
+    Application.get_env(@app, {:format, encoding_name}, %__MODULE__{encoding_name: encoding_name})
```

Comment:
Shouldn't we return here empty map or nil? I mean doc says `registered` but if get_env returns default value it means given `encoding_name` isn't registered so we probably should return `%{}/nil`

Reply:
The format can be also 'partially registered', for example, there's a payloader but not a depayloader. In other words, unknown fields are always set to `nil`. Thus, returning some custom value for unregistered format would make that inconvenient to interact with.